Update: support forward auth for dashboard

till opened the pull request:
## forward auth / external authentication

 - enable with `traefik_dashboard_fwdauth_server` (`https://fqdn`)
 - append forward auth labels to traefik (dashboard) service when necessary
 - added guards to ensure things are configured correctly (e.g. basic auth or forward auth)

## traefik dashboard

 - introduce `{{ traefik_dashboard_fqdn }}`
 - allow setting the complete fqdn of the traefik dashboard
 - defaults to `{{ traefik_dashboard_subdomain }}.{{ inventory_hostname }}`

till committed:
Update: support forward auth for dashboard

 - enable with traefik_dashboard_fwdauth_server
 - added guards to ensure things are configured
 - append fwdauth labels to service when necessary

rwos reviewed (approved):
LGTM, surprisingly easy 😄

till commented:
Note to myself: deploy this somewhere before I go through a cycle of tag/fix/tag/fix.

till commented on vars/main.yml:
```diff
@@ -1,7 +1,12 @@
   - "traefik.port={{ traefik_dashboard_entrypoint_port }}"
   - "traefik.docker.network={{ traefik_docker_network }}"
   - "traefik.backend=traefik"
   - "traefik.frontend.rule=Host:{{ traefik_dashboard_subdomain }}.{{ traefik_docker_domain | default(inventory_hostname) }}"
```
I probably need to add another domain here.

till committed:
Update: allow setting the fqdn

By default "traefik.{{ inventory_hostname }}". This new config var
allows setting the complete name to something else if needed.

We added this to unify all dashboards on a host to be able to fwd
auth everywhere.

till committed:
Update: allow overriding deps [this is ugly]

till committed:
Chore: improve readme + shameless self promo :)

rwos commented on defaults/main.yml:
```diff
@@ -16,11 +16,13 @@ traefik_docker_network: "public"
 
 # Docker
-# traefik_docker_domain: "mydomain.org" (defaults to inventory_hostname)
+traefik_docker_domain: "{{ inventory_hostname }}"
```
Does that work with the order of the variables? Like, how this is defined after it is used in `traefik_dashboard_fqdn` (it could work, I don't know).

till replied:
@rwos I think I'll add a test to make sure. 😂

rwos reviewed (approved):
LGTM, except unsure if the vars stuff works (see comment), but if it does - 👍
